reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscription.scala

```diff
@@ -67,57 +68,90 @@ private[reactivestreams] final class StreamSubscription[F[_], A](
         go(in).stream
       }
 
-    val s =
+    val prog =
       stream
         .through(subscriptionPipe)
-        .interruptWhen(cancelled)
+        .interruptWhen(canceled)
         .evalMap(x => F.delay(sub.onNext(x)))
         .handleErrorWith(e => Stream.eval(onError(e)))
-        .onFinalize(cancelled.get.ifM(ifTrue = F.unit, ifFalse = onComplete))
+        .onFinalize(canceled.get.ifM(ifTrue = F.unit, ifFalse = onComplete))
         .compile
         .drain
```

Comment:
Staring at this some more, I wonder if it should be:
```scala
      stream
        .through(subscriptionPipe)
        .foreach(x => F.delay(sub.onNext(x)))
        .compile
        .drain
        .guaranteeCase {
           case Success(...) => onComplete
           case Error(...) => onError
           case Canceled => onCancel
         }
        .race(canceled.get)
```
Then, I wonder if `canceled` can just be a `Deferred[Unit]` 🤔